src/uicomponents/LinkButton.tsx

```diff
@@ -14,10 +14,44 @@ import { RaidBattleInfo } from "../raidcalc/RaidBattle";
 
 import PokedexService from "../services/getdata";
 import { MoveData, RaidTurnInfo, SubstituteBuildInfo, TurnGroupInfo } from "../raidcalc/interface";
-import { getTranslation } from "../utils";
+import { encode, decode, getTranslation } from "../utils";
 
+import { db } from "../config/firestore";
+import { collection, doc, getCountFromServer, getDoc, setDoc } from "firebase/firestore";
+import STRAT_LIST from "../data/strats/stratlist.json";
 
 const gen = Generations.get(9);
+const JSON_HASHES = Object.entries(STRAT_LIST).map(([boss, strats]) => Object.entries(strats as Object).map(([name, h]) => h as string)).flat();
+
+async function getFullHashFromShortHash(hash: string): Promise<string> {
+    let cleanHash = hash;
+    if (hash[0] === "#") {
+        cleanHash = hash.slice(1);
+    }
+    const hashID = decode(cleanHash);
+    const docRef = doc(db, "links", `${hashID}`);
+    const docSnap = await getDoc(docRef);
+    if (!docSnap.exists()) {
+      console.log("Invalid link");
+    }
+    const data = docSnap.data();
+    return data ? data.hash : null;
+}
+
+async function generateShortHash(): Promise<[number, string]> {
```

Comment:
Would it be possible for a collision here? If someone happens to get a short hash that collides with the name of one of our saved JSON hashes, would they be unable to access their data because the app will choose to load the one from STRAT_LIST? I don't think this is a realistic issue that will ever come up though.

Reply:
If this is an issue a quick solution is to add an underscore to the end of every STRAT_LIST hash since that's not in the alphabet for possible short hash characters.

Reply:
This is certainly possible. I think there are only 6 potential collisions so far, and 6 / 62^8 is rare enough that I'm not too worried about it. But your suggestion is a good one, and it would be an easy fix to make.

If we don't want to break old links, now that collisions are checked before writing to the database, another option would be to manually add entries in the database for all of the decoded boss names in STRAT_LIST as a preventative measure.